patroni/postgresql/config.py

```diff
@@ -410,6 +419,7 @@ def write_postgresql_conf(self, configuration: Optional[CaseInsensitiveDict] = N
             configuration['hot_standby_feedback'] = 'on'
 
         with ConfigWriter(self._postgresql_conf) as f:
+            self.set_file_permissions(self._postgresql_conf)
```

Comment:
Just a side thought...

I was wondering if somehow we could use `os.fchmod` inside `ConfigWriter.__enter__` to set permissions of the file descriptor as we are always calling `self.set_file_permissions` right after entering the context.

However that class is not "PGDATA aware", and we would somehow copy the logic of `set_file_permissions`, so...